### PR TITLE
Add originalPrice to product response

### PR DESCRIPTION
**Pull request by alexander-schranz**
I think the originalPrice is a information which is also mostly needed when reading products.

### CODE REVIEW

File: spec/Factory/Product/ProductVariantViewFactorySpec.php [hunk 1]
```diff
@@ -80,6 +80,7 @@ function it_builds_product_variant_view(
         $channel->getBaseCurrency()->willReturn($currency);
         $currency->getCode()->willReturn('PLN');
         $channelPrice->getPrice()->willReturn(500);
+        $channelPrice->getOriginalPrice()->willReturn(null);
```
Comment (mamazu) [inline spec/Factory/Product/ProductVariantViewFactorySpec.php]: Should also be tested with originalPrice != null

Comment (alexander-schranz) [reply spec/Factory/Product/ProductVariantViewFactorySpec.php]: @mamazu test added

File: src/Factory/Product/ProductVariantViewFactory.php [hunk 2]
```diff
@@ -42,6 +42,13 @@ public function create(ProductVariantInterface $variant, ChannelInterface $chann
             $channel->getBaseCurrency()->getCode()
         );
 
+        if ($originalPrice = $channelPricing->getOriginalPrice()) {
```
Comment (mamazu) [inline src/Factory/Product/ProductVariantViewFactory.php]: Implicit integer to boolean cast (not type safe)

File: src/Factory/Product/ProductVariantViewFactory.php [hunk 3]
```diff
@@ -42,7 +42,8 @@ public function create(ProductVariantInterface $variant, ChannelInterface $chann
             $channel->getBaseCurrency()->getCode()
         );
 
-        if ($originalPrice = $channelPricing->getOriginalPrice()) {
+        $originalPrice = $channelPricing->getOriginalPrice()
+        if (null !== $originalPrice) {
```
Comment (mamazu) [inline src/Factory/Product/ProductVariantViewFactory.php]: The behavior you previously encoded was `$originalPrice !== 0`.

Comment (alexander-schranz) [reply src/Factory/Product/ProductVariantViewFactory.php]: If the originalPrice is 0 it should be returned as 0 and only if really not set it should not be returned or?

Comment (mamazu) [reply src/Factory/Product/ProductVariantViewFactory.php]: Is it possible for the original price to be null?

Comment (alexander-schranz) [reply src/Factory/Product/ProductVariantViewFactory.php]: Yes if not set its null

Comment (mamazu) [reply src/Factory/Product/ProductVariantViewFactory.php]: Okay then it okay.

**Comment by mamazu** [comment]
Thank you, Alexander! :tada:

File: tests/Responses/Expected/order/order_details_response.json [hunk 4]
```diff
@@ -33,6 +33,10 @@
                             "current": 1999,
                             "currency": "GBP"
                         },
+                        "originalPrice": {
+                            "current": 2999,
+                            "currency": "GBP"
```
Comment (lchrusciel) [inline tests/Responses/Expected/order/order_details_response.json]: Is there any possibility to add original price in a different currency than the current one? What would you say, about merging these `originalPrice` and `price` fields?

Comment (mamazu) [reply tests/Responses/Expected/order/order_details_response.json]: I think both should be used with the same structure. The price is renderd using a `Priceview` so should the original price be otherwise it would make things more difficult when consuming this API.
